src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/NameQuotedValuePairListEncoding.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.Designer;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug
+{
+
+    [Export(typeof(INameValuePairListEncoding))]
+    [ExportMetadata("Encoding", "NameQuotedValuePairListEncoding")]
+    internal sealed class NameQuotedValuePairListEncoding : INameValuePairListEncoding
+    {
+        public IEnumerable<(string Name, string Value)> Parse(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                yield break;
+            }
```

Comment:
Consider:

```suggestion
            Requires.NotNullOrWhiteSpace(value, nameof(value));
```

The rationale here is that a `null` value would be an error (the argument is non-nullable), and throwing when we have invalid values helps to discover issues, rather than failing silently. If we did fail silently here, we might return an empty sequence which would then potentially also overwrite customer data (assuming the issue happened between getting the value from the project and passing it to this class).

Equally, an empty or white space string cannot be parsed correctly, so should throw.